kv/dist_sender.go

```diff
@@ -650,6 +656,7 @@ func (ds *DistSender) Send(_ context.Context, call proto.Call) {
 
 	curReply := finalReply
 	for {
+		ds.tracer.Trace(args.Header(), "ds_send")
```

Comment:
Is it legal to call Trace without Begin? I think I'd like to see all Trace calls happen within a Begin/Finalize pair (this would give them a place to exist in a timeline-based UI). Maybe Begin() should return an object with Trace and Finalize methods instead of making all 3 methods on Tracer (this way the args.Header() parameter wouldn't need to be repeated).